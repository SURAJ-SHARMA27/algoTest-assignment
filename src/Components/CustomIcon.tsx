import React from 'react';

const CustomIcon = () => {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      width="18" // Set your desired width
      height="18" // Set your desired height
      viewBox="0 0 24 24"
      fill="none" // Set fill color if needed
      stroke="#CA8144" // Use current color for stroke
      strokeWidth="2" // Set stroke width
      strokeLinecap="round"
      strokeLinejoin="round"
    >
      <path d="M12 9v3.75m-9.303 3.376c-.866 1.5.217 3.374 1.948 3.374h14.71c1.73 0 2.813-1.874 1.948-3.374L13.949 3.378c-.866-1.5-3.032-1.5-3.898 0L2.697 16.126zM12 15.75h.007v.008H12v-.008z"></path>
    </svg>
  );
};

export default CustomIcon;
