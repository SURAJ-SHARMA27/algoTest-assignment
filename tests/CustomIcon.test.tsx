import { render, screen } from '@testing-library/react';
import CustomIcon from '../src/Components/CustomIcon';
import { it, expect, describe } from 'vitest';
import React from 'react';
import '@testing-library/jest-dom/vitest';

describe('CustomIcon Component', () => {
  it('renders the CustomIcon component without crashing', () => {
    render(<CustomIcon />);
    const svgElement = screen.getByTestId('customIcon');
    expect(svgElement).toBeInTheDocument();
  });

  it('matches the snapshot', () => {
    const { asFragment } = render(<CustomIcon />);
    expect(asFragment()).toMatchSnapshot();
  });


  it('has correct width and height', () => {
    render(<CustomIcon />);
    const svgElement = screen.getByTestId('customIcon');
    expect(svgElement).toHaveAttribute('width', '18');
    expect(svgElement).toHaveAttribute('height', '18');
  });

  it('has the correct stroke color', () => {
    render(<CustomIcon />);
    const svgElement = screen.getByTestId('customIcon');
    expect(svgElement).toHaveAttribute('stroke', '#CA8144');
  });
});
