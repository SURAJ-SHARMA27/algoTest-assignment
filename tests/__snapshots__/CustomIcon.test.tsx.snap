// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CustomIcon Component > matches the snapshot 1`] = `
<DocumentFragment>
  <svg
    data-testid="customIcon"
    fill="none"
    height="18"
    stroke="#CA8144"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="18"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M12 9v3.75m-9.303 3.376c-.866 1.5.217 3.374 1.948 3.374h14.71c1.73 0 2.813-1.874 1.948-3.374L13.949 3.378c-.866-1.5-3.032-1.5-3.898 0L2.697 16.126zM12 15.75h.007v.008H12v-.008z"
    />
  </svg>
</DocumentFragment>
`;
